WooCommerce/Classes/ViewRelated/Dashboard/TopPerformers/TopPerformersDashboardViewModel.swift

```diff
@@ -103,13 +137,79 @@ private extension TopPerformersDashboardViewModel {
         let action = AppSettingsAction.setLastSelectedStatsTimeRange(siteID: siteID, timeRange: timeRange)
         stores.dispatch(action)
     }
+
+    func updateResultsController() {
+        let resultsController = createResultsController(timeRange: timeRange)
+        self.resultsController = resultsController
+        resultsController.onDidChangeContent = { [weak self] in
+            self?.updateUIInLoadedState()
+        }
+        resultsController.onDidResetContent = { [weak self] in
+            self?.updateUIInLoadedState()
+        }
+
+        do {
+            try resultsController.performFetch()
+        } catch {
+            ServiceLocator.crashLogging.logError(error)
+        }
+    }
+
+    func updateUIInLoadingState() {
+        periodViewModel.update(state: .loading)
+    }
+
+    func updateUIInLoadedState() {
+        guard !syncingData else {
+            return
+        }
```

Comment:
❓ non-blocking: Trying to understand this part.

When does it get called where `syncingData` might still be true? Is it to handle things like multiple pull-to-refresh actions?

Reply:
This can be called when we get data from Core data (in line 145 or 148). When updating the time range, I call `updateResultsController` and `reloadData` so if we get the result from core data before `reloadData` finishes, we can reach this case. I haven't tested but it's very likely to happen IMO.